lib/cn/vblock_builder.c

```diff
@@ -172,6 +174,10 @@ vblock_start(struct vblock_builder *bld, const struct key_obj *min_kobj)
     key_obj_copy(bld->cur_minkey, sizeof(bld->cur_minkey), &bld->cur_minklen, min_kobj);
 
     return 0;
+
+err_exit:
+    bld->destruct = true;
+    return err;
```

Comment:
Couldn't you just set bld->destroy to true at the top, then clear it on successful exit?  Then you don't need any goto's... nor this new chunk at the end...

Reply:
Good idea, let me try that out.

Reply:
Done, thanks!